.gitignore

```diff
@@ -3,9 +3,13 @@
 
 ### Files private to developers
 
+# Files that should be git-ignored, but are hand-edited or otherwise valued,
+# and so should not be destroyed by "make clean".
+# start-noclean
 requirements/private.txt
 lms/envs/private.py
 cms/envs/private.py
+# end-noclean
```

Comment:
It is certainly possible that other files should be protected. I don't use those IDEs, so I don't know what makes sense. I'm not sure what those people do today to protect those files.

This code will handle multiple start-noclean/end-noclean sections, so we can add more comments to .gitignore for those files.